src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -408,12 +408,16 @@ private static function remove_json_comments( $array ) {
 	 * @return array Custom Post Type for the user's origin config.
 	 */
 	public static function get_user_data_from_wp_global_styles( $theme, $create_post = false, $post_status_filter = array( 'publish' ) ) {
-		if ( ! static::theme_has_support() ) {
-			return array();
-		}
 		if ( ! $theme instanceof WP_Theme ) {
 			$theme = wp_get_theme();
 		}
+
+		// Bail early if the theme does not support a theme.json. Since WP_Theme_JSON_Resolver::theme_has_support()
+		// only supports the active theme, the extra condition for whether $theme is the active theme is present here.
+		if ( $theme->get_stylesheet() === get_stylesheet() && ! static::theme_has_support() ) {
```

Comment:
Instead of doing this, how about we add a method to the WP_Theme class. So we could do this, 
```
if( $theme->theme_has_support() ) {
```

We could even use the `get_file_path` method of the WP_Theme class.

Reply:
I have worked on a moving this to the WP_Theme class. See https://github.com/spacedmonkey/wordpress-develop/commit/b10228fc4aa106438e0a989f30b7340973d8f786. This change is inline with `is_block_theme` and `is_block_theme` method on WP_theme.

Reply:
@spacedmonkey I like that idea, however I think this goes beyond the scope of what would fit into a WordPress patch release. I think adding API methods/functions should be part of a regular WP release like 6.2.

Reply:
The problem is, I don't love this line. 

```
if ( $theme->get_stylesheet() === get_stylesheet() && ! static::theme_has_support() ) {
```

It is only relates to current theme. My change would fix that issue.

Reply:
At the very least, we should not use wp_get_theme that creates a new object each time.

Reply:
I saw you opened a follow up ticket, SGTM!